disnake/ext/commands/converter.py

```diff
@@ -343,9 +344,10 @@ async def convert(self, ctx: AnyContext, argument: str) -> disnake.User:
                 except disnake.HTTPException:
                     raise UserNotFound(argument) from None
 
-            if isinstance(result, disnake.Member):
+            if not isinstance(result, disnake.User):
                 return result._user
-            return result  # type: ignore
+
+            return result
```

Comment:
For some reason, pyright did not want to narrow on the 2nd return that `result` could not be `Member` (>ლ)